scarb/src/ops/package.rs

```diff
@@ -205,17 +208,45 @@ fn list_one_impl(
     ws: &Workspace<'_>,
 ) -> Result<Vec<Utf8PathBuf>> {
     let pkg = ws.fetch_package(&pkg_id)?;
-    let recipe = prepare_archive_recipe(pkg, opts)?;
+    let recipe = prepare_archive_recipe(pkg, opts, ws)?;
     Ok(recipe.into_iter().map(|f| f.path).collect())
 }
 
-fn prepare_archive_recipe(pkg: &Package, opts: &PackageOpts) -> Result<ArchiveRecipe> {
+fn get_crate_archive_basename(cargo_toml_path: Utf8PathBuf) -> String {
```

Comment:
I don't think we need to parse the Cargo manifest at all. We could just take the name and version from package directly - it's assumed Cargo name and version matches this.

Reply:
> it's assumed Cargo name and version matches this.

I don't think we should assume that. That's not true in in case `snforge_scarb_plugin` (`snforge-scarb-plugin` in Cargo.toml), that's not true in case of plugin generated by project builder, at least in the current state (so I would guess it hasn't been an assumption before).

Sure, we might *expect* that, but people will still make mistakes and errors will get thrown. The error won't be too intuitive as it will relate to inability resolve / open certain path. The only way to generate human-readable error is to parse Cargo manifest and compare it against Scarb manifest. Which brings us to the same thing we wanted to get rid of.

> This function cannot be used with unnormalized manifest. 

Great catch. 
This a little problematic though, as we use `get_crate_archive_basename` to determine path to normalized Cargo.toml. 

I suppose we'd have to extract version from `cargo metadata`. Unless you have any better ideas in mind?

Reply:
> I don't think we should assume that. 

Yep, I stand corrected, thanks! What we do assume, is that if you replace `-` with `_` in Cargo package name, it is the same as Scarb package. 

Actually I am inclined to think we should just enforce both  name and version to be the same in Scarb.toml and Cargo.toml. I don't really see why shouldn't we, wdyt?

Reply:
> Actually I am inclined to think we should just enforce both name and version to be the same in Scarb.toml and Cargo.toml. I don't really see why shouldn't we, wdyt?

I think it there are good reasons to keep allowing different package names across Scarb and Cargo manifests, as discussed offline:
- Changing this would be breaking to already existing packages
- There are already existing different naming conventions: `-` and `_` should be used for Cargo and Scarb respectively
- We would need to have checks anyway in order to produce human-readable errors, so we wouldn't be simplifying any logic anyway

For the latter reason, I [updated](https://github.com/software-mansion/scarb/pull/1605#discussion_r1780209827
) the way name and version are extracted. 

We can discuss this further and potentially make go through with enforcing package name equality, but as it stands I'm not sure we're gaining anything from it, while having slightly negative impact on user QoL.

Reply:
Ok, but please emit some warning saying there is a mismatch :)

Reply:
Addressed in [c5d6cff](https://github.com/software-mansion/scarb/pull/1605/commits/c5d6cff80b39abdcf62adc76b7ebf7f8acff9dac)